core/field_image.js

```diff
@@ -124,6 +122,47 @@ Blockly.FieldImage.prototype.EDITABLE = false;
  */
 Blockly.FieldImage.prototype.isDirty_ = false;
 
+/**
+ * Configure this field's customization options. Taking into account old
+ * parameter ordering on the constructor.
+ * @param {Function|string|null} opt_onClick Option function to be called
+ *    when the image is clicked. Or alt text.
+ * @param {Object|Function|null} opt_config A map of options used to
```

Comment:
are these all optional at this point? if not, rename opt_

Reply:
Sort of optional? You can pass null just fine. I think it's confusing because this function is handling backwards compatibility for a different function.

Option 1:
Move the backwards compat code back to the constructor. Handle only configuration inside config_.

Option 2:
Come up with a different naming scheme for this particular config_ function. Currently I do not have any ideas.

Option 3:
Leave as is.

Reply:
Why not keep the constructor parameters as is, and just provide a way to override them in opt_config. 
I'm wondering why the decision was made to move rtl and alt into config, but everything else is still there?
What makes rtl a configuration option more than say something like ``onClick``?